src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { Env } from 'src/env/index';
+import { JwtPayload } from './jwt-interface';
+import { RedisService } from '@liaoliaots/nestjs-redis';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
+  constructor(
+    private readonly redisService: RedisService,
+  ) {
+    super({
+      secretOrKey: Env.JWT.AUTH_TOKEN,
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+    });
+  }
+
+  async validate(payload: JwtPayload) {
+    const { userId } = payload;
+    const client = this.redisService.getClient();
+    const jwtToken = await client.get(`jwt:${userId}`);
+    if (!jwtToken) {
+      throw new UnauthorizedException();
+    }
+    return payload;
```

Comment:
PullProofAI

It's recommended to add comments to explain the purpose of the `validate` method and each step within it for better code understanding.
```diff
+   // Validate the JWT payload
+   async validate(payload: JwtPayload) {
+     // Extract userId from payload
+     const { userId } = payload;
+     // Get Redis client
+     const client = this.redisService.getClient();
+     // Retrieve JWT token from Redis
+     const jwtToken = await client.get(`jwt:${userId}`);
+     // Check if JWT token exists
+     if (!jwtToken) {
+       throw new UnauthorizedException();
+     }
+     // Return the payload if validation succeeds
+     return payload;
+   }
```



<!-- This is an auto-generated comment by PullProofAI -->